### PR TITLE
fix: issue with document highlighting offset

### PR DESCRIPTION
**Pull request by broulaye**
#### What do these changes do/fix?
This changes fixes the highlighting of the attributes in CIDocument, you can see it's working properly by running storybook

#### How do you test/verify these changes?
 run yarn storybook and navigating to CIDocument -> default. select defined terms under attributes you should see that everything is highlighted correctly.

#### Have you documented your changes (if necessary)?
Yes
#### Are there any breaking changes included in this pull request?
Yes

### CODE REVIEW

File: packages/discovery-react-components/src/utils/document/documentUtils.ts [hunk 1]
```diff
@@ -84,7 +85,26 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
     throw new Error(`Failed to find text node. Node: ${node.textContent}, offset: ${offset}`);
   }
 
-  const textOffset = Math.max(0, offset - runningOffset);
+  let textOffset = Math.max(0, offset - runningOffset);
+
+  // If attribute 'data-orig-text' is present then
```
Comment (maniax89) [inline packages/discovery-react-components/src/utils/document/documentUtils.ts]: do we have a test for this?

File: packages/discovery-react-components/src/utils/document/processDoc.ts [hunk 2]
```diff
@@ -266,6 +267,21 @@ function setupSectionParser(
     },
 
     ontext: (_: Parser, text: string): void => {
+      // In order to properly highlight texts in the DOM we need
+      // to know if the original text contains some encoded
+      // html entities, and if so, we pass that value down to
+      // be used later.
+      if (decodeHTML(text).length != text.length) {
```
Comment (maniax89) [inline packages/discovery-react-components/src/utils/document/processDoc.ts]: do we have a test for this?

Comment (jhpedemonte) [reply packages/discovery-react-components/src/utils/document/processDoc.ts]: Added a test.

File: packages/discovery-react-components/src/utils/document/documentUtils.ts (hunk 1 above)
Comment (jhpedemonte) [reply packages/discovery-react-components/src/utils/document/documentUtils.ts]: I took a stab, but there are some differences between browser DOM and `jsdom` (as used in tests). I'll make an issue to add some tests around this.

**Pull request merged**